### PR TITLE
✅  Add tests for PageExperienceCheck.js

### PR DESCRIPTION
**Pull request by robinvanopstal**

### CODE REVIEW

File: pixi/src/checks/PageExperienceCheck.test.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+beforeAll(() => {
+  jest.spyOn(PageExperienceCheck.prototype, 'fetchJson').mockImplementation(() => {
+    return dummyApiResponse
+  });
+});
+
+describe('Page experience check', () => {
+  test('create report data for url', async () => {
```
Comment (sebastianbenz) [inline pixi/src/checks/PageExperienceCheck.test.js:16]: nit: when using `describe` combine it with `it`. Ideally, the describe block creates a meaningful sentence when combined with the `it` block: 'Page experience check fetches report data from PSI'.

Comment (sebastianbenz) [reply pixi/src/checks/PageExperienceCheck.test.js:16]: It's also fine to use `test` without a describe block. The filename serves as a categorization in that case.

File: pixi/src/checks/PageExperienceCheck.test.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+beforeAll(() => {
+  jest.spyOn(PageExperienceCheck.prototype, 'fetchJson').mockImplementation(() => {
```
Comment (sebastianbenz) [inline pixi/src/checks/PageExperienceCheck.test.js:10]: I'd mock on a fetch level instead. For example using `fetch-mock` which we're already using in the project.

File: pixi/src/checks/PageExperienceCheck.test.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    return dummyApiResponse
```
[the remark this answers is not in the document]
Comment (robinvanopstal) [reply pixi/src/checks/PageExperienceCheck.test.js:11]: Will be solved with https://github.com/ampproject/amp.dev/pull/4313#discussion_r472029451

**Comment by matthiasrohmer** [comment]
@googlebot I consent.

**Comment by matthiasrohmer** [comment]
@robinvanopstal, can you please mark this as ready to review? Thanks!

**Review by sebastianbenz — APPROVED** [review]

**Comment by matthiasrohmer** [comment]
@sebastianbenz, can you please have another look? https://github.com/ampproject/amp.dev/pull/4313/commits/6af3d37a1470fd3e6bf235b5738cf0ab54c111bd illustrates how it could work...

**Review by sebastianbenz — APPROVED** [review]
Nice! Thanks!

**Comment by sebastianbenz** [comment]
Are these test already run as part of `npm test`?

**Comment by matthiasrohmer** [comment]
Yes they should be as `npm test` executes all scripts prefixed with `test:` 🙂

**Pull request merged**